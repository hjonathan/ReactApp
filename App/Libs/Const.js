import RNFS from "react-native-fs";

export const CASE = {
    INBOX: "Inbox",
    DRAFT: "Draft",
    PARTICIPATED: "Participated",
    UNASSIGNED: "Unassigned",
    STATUS: {
        NONE: "none",
        SENDING: "sending",
        WORKING: "working",
        OFFLINE: "offline"
    }
};

export const DATETIME = {
    LOCAL: "YYYY-MM-DD HH:mm:ss",
    LOCAL_UTC: "YYYY-MM-DD HH:mm:ss Z"
};

export const DATE = {
    SIMPLE: "YYYYMMDD",
    LOCAL: "YYYY-MM-DD"
};

export const TIME = {
    LOCAL: "HH:mm:ss"
};

export const DS = "/";

export const PATH = {
    PROJECT: `${RNFS.DocumentDirectoryPath}`,
    FORMRENDER: `${DS}FormRender`,
    STEPS: `${DS}Steps`,
    NEXTSTEPS: `${DS}NextSteps`,
    FROMS: `${DS}Forms`,
    DATA: `${DS}Data`,
    EXTERNALLIBS: `${DS}ExternalLibs`,
    BUILDPROD: `${DS}build-prod`,
    MUSIC: "/storage/emulated/0/Music/processmakeraudiorecord/"
};

export const FILENAMES = {
    EXTERNALMAP: "ExtMap.json"
};
