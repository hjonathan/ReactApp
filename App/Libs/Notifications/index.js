/* eslint-disable import/prefer-default-export */
import * as pushNotification from "./pushNotification";

export {pushNotification};
