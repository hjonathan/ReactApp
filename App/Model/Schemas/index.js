import Realm from "realm";
import CounterSchema from "./Counter";
import * as Lists from "./List";
import TodoSchema from "./Todo";
import DraftSchema from "./Draft";
import ParticipatedSchema from "./Participated";
import UnassignedSchema from "./Unassigned";
import StartCaseSchema, {StartCaseStepSchema} from "./StartCase";
import UserSchema from "./User";
import StepSchema from "./Step";
import StepTriggerSchema from "./StepTrigger";
import FormSchema from "./Form";
import NewCaseSchema from "./NewCase";
import DataSchema from "./Data";
import MobileControlSchema from "./MobileControl";

/**
 * Return the Realm instance
 * @type {Realm}
 */
export const RealmObject = new Realm({
    schema: [
        UserSchema,
        CounterSchema,
        Lists.ProcessSchema,
        Lists.TaskSchema,
        Lists.UserOwnerSchema,
        Lists.UsersSchema,
        TodoSchema,
        DraftSchema,
        ParticipatedSchema,
        UnassignedSchema,
        StartCaseSchema,
        StartCaseStepSchema,
        StepSchema,
        StepTriggerSchema,
        FormSchema,
        NewCaseSchema,
        DataSchema,
        MobileControlSchema
    ],
    schemaVersion: 1.0
});

/**
 * Return all Schemas
 */
export {
    UserSchema,
    CounterSchema,
    TodoSchema,
    DraftSchema,
    ParticipatedSchema,
    UnassignedSchema,
    StartCaseSchema,
    StepSchema,
    FormSchema,
    NewCaseSchema,
    DataSchema,
    MobileControlSchema
};
