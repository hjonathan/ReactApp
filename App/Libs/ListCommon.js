import _ from "lodash";
import Task from "../Model/Task";
import Step from "../Model/Step";
import {CASE} from "./Const";

/**
 * Function commons for: inbox, draft, unassigned and participated
 */
export default class ListCommon {
    /**
     * Constructor
     * @param props
     */
    constructor (props) {
        this.props = props;
        this.task = Task.allTaskIdOffline();
    }

    /**
     * Get data lists
     * @param method
     * @param params
     */
    getDataList (method, params = {}) {
        if (this.props[method]) {
            this.props[method](params);
        }
    }

    /**
     * Gets usersInfo
     * @param caseList
     * @param type is type in list: prevUser, currentUser
     */
    getUsersInfo (caseList, type) {
        if (caseList.length > 0 && caseList[0][type]) {
            const ids = caseList.map(cases => cases[type].userId);
            this.props.userInfoRequest({
                size: "1",
                user: {
                    ids: _.uniq(ids)
                }
            });
        }
    }

    /**
     * Merges userPhoto props to dataList
     * @param params
     * @param type is type in list: prevUser, currentUser
     */
    mergeList (params, type) {
        let i,
            j,
            cbUser = function (user) { return user.userId === params.caseList[i][type].userId; };
        if (params.caseList) {
            for (i = 0; i < params.caseList.length; i += 1) {
                if (params.caseList[i][type]) {
                    j = _.findIndex(params.users, cbUser);
                    if (j >= 0) {
                        params.caseList[i].userPhoto = params.users[j].userPhoto;
                    }
                    params.caseList[i].status = this.processIfOffline(params.caseList[i]);
                }
            }
        }
        return params.caseList;
    }

    /**
     * Check if case is offline and get all necessary for case
     * @param caseData
     * @returns {boolean}
     */
    processIfOffline (caseData) {
        let isOffline,
            {status} = caseData;
        if (!status || status === CASE.STATUS.NONE) {
            if (caseData.task.offlineEnabled === "TRUE" && this.props.caseOffline) {
                this.props.caseOffline(caseData);
                status = CASE.STATUS.OFFLINE;
            }
            isOffline = Step.existSteps(caseData.process.processId, caseData.task.taskId);
            if (isOffline && _.includes(this.task, caseData.task.taskId)) {
                status = CASE.STATUS.OFFLINE;
            }
        }
        return status;
    }

    /**
     * Key for FlatList formed by delIndex and caseId
     * @param item
     * @returns {string}
     */
    keyExtractor (item, index) {
        return `${item.delIndex}_${item.caseId}`;
    }
}
