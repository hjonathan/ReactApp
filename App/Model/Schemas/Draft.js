import * as List from "./List";
import {CASE} from "../../Libs/Const";

/**
 * Schema for model Draft list
 */
export default class DraftSchema {
    static schema = {
        name: "DraftSchema",
        primaryKey: "caseId",
        properties: {
            caseId: {type: "string"},
            delIndex: {type: "string"},
            user: {type: List.UserOwnerSchema.schema.name},
            currentUser: {type: List.UsersSchema.schema.name},
            task: {type: List.TaskSchema.schema.name},
            process: {type: List.ProcessSchema.schema.name},
            caseNumber: {type: "string"},
            caseTitle: {type: "string"},
            date: {type: "string?"},
            delegateDate: {type: "string?"},
            dueDate: {type: "string"},
            status: {type: "string?", default: CASE.STATUS.NONE}
        }
    };
}
