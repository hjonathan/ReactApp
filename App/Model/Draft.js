import _ from "lodash";
import BaseModel from "./Interface/BaseModel";
import {RealmObject, DraftSchema, DataSchema} from "./Schemas";
import {CASE} from "../Libs/Const";

/**
 * Model for list Draft
 */
class DraftCase extends BaseModel {
    constructor () {
        super(DraftSchema.schema.name);
    }

    /**
     * Save the list when the action cases/draft/success is executed
     * @param listDraft
     */
    saveList (listDraft) {
        let dbData;
        RealmObject.write(() => {
            listDraft.data.response.map((draftCase) => {
                let otherCase = RealmObject.objects(DataSchema.schema.name).filtered(`caseId = "${draftCase.caseId}" AND type != "${CASE.DRAFT}"`);
                RealmObject.delete(otherCase);
                dbData = _.values(RealmObject.objects(DataSchema.schema.name).filtered(`caseId = "${draftCase.caseId}" AND type = "${CASE.DRAFT}"`)).shift();
                if (dbData) {
                    draftCase.status = dbData.status;
                    RealmObject.create(DraftSchema.schema.name, draftCase, true);
                    return;
                }
                RealmObject.create(DraftSchema.schema.name, draftCase, true);
                return;
            });
        });
    }

    /**
     * Return all rows of list Draft
     */
    getDraftOffline () {
        let result = RealmObject.objects(DraftSchema.schema.name).sorted("delegateDate", true);
        return result.map(x => _.assign({}, x));
    }

    /**
     * Return all rows in status sending
     * @returns {any[]}
     */
    getDraftSending () {
        let result = RealmObject.objects(DraftSchema.schema.name).filtered(`status = "${CASE.STATUS.SENDING}"`);
        return result.map(x => _.assign({}, x));
    }

    /**
     * Return a rows of case Draft
     */
    getDraftCase (caseId) {
        return _.values(RealmObject.objects(DraftSchema.schema.name).filtered(`caseId = "${caseId}"`)).shift();
    }

    /**
     * Check if a cases exist in Draft
     * @param caseId
     * @returns {boolean}
     */
    existCase (caseId) {
        const result = RealmObject.objects(DraftSchema.schema.name).filtered(`caseId = "${caseId}"`);
        return result.length > 0;
    }

    /**
     * Change status of row
     * @param caseId
     * @param status
     */
    changeStatus (caseId, status) {
        let dbDraft;
        RealmObject.write(() => {
            dbDraft = _.values(RealmObject.objects(DraftSchema.schema.name).filtered(`caseId = "${caseId}"`)).shift();
            if (dbDraft) {
                dbDraft.status = status;
                RealmObject.create(DraftSchema.schema.name, dbDraft, true);
            }
        });
    }

    /**
     * Remove the specified data case of storage.
     * @param value
     */
    destroyCase (value) {
        RealmObject.write(() => {
            let data = RealmObject.objects(DraftSchema.schema.name).filtered(`caseId = "${value}"`);
            RealmObject.delete(data);
        });
    }

    /**
     * Delete multiple Case in Draft
     */
    destroyObject () {
        RealmObject.write(() => {
            let allObjects = RealmObject.objects(DraftSchema.schema.name);
            RealmObject.delete(allObjects); // Deletes all case in Draft
        });
    }
}

export default new DraftCase();
