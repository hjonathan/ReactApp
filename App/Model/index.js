import Step from "./Step";
import User from "./User";
import StartCase from "./StartCase";
import Todo from "./Todo";
import Unassigned from "./Unassigned";
import Participated from "./Participated";
import Draft from "./Draft";
import Form from "./Form";
import NewCase from "./NewCase";
import Data from "./Data";
import Task from "./Task";
import MobileControl from "./MobileControl";

/**
 * Returns the models
 */
export {
    Step,
    User,
    StartCase,
    Todo,
    Unassigned,
    Participated,
    Draft,
    Form,
    NewCase,
    Data,
    Task,
    MobileControl
};
