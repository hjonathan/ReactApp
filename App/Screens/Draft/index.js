import {connect} from "react-redux";
import Draft from "./draft";
import actions from "../../Actions";
import {CASE} from "../../Libs/Const";

const mapStateToProps = state => ({
        cases: {
            draft: state.Cases.draft,
            users: state.Cases.users
        },
        list: CASE.DRAFT,
        forceRefresh: state.Cases.forceRefresh || false,
        disableItemList: state.Screens.All.disableItemList,
        isConnected: state.Net.isConnected
    }),
    mapDispatchToProps = dispatch => ({
        requestCasesDraft: params => dispatch(actions.cases.draft.request(params)),
        userInfoRequest: params => dispatch(actions.cases.userInfo.request(params)),
        caseOffline: params => dispatch(actions.net.case.list(params)),
        routeName: name => dispatch(actions.cases.routeName(name)),
        reset: () => dispatch(actions.cases.reset())
    });

export default connect(mapStateToProps, mapDispatchToProps)(Draft);
