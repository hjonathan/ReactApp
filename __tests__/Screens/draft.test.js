import React from 'react'
import {RealmObject} from "../../App/Model/Schemas";
import {shallow} from 'enzyme';
import Draft from '../../App/Screens/Draft';
import DraftScreen from '../../App/Screens/Draft/draft';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import {createStore} from 'redux';
import {CASE} from "../../App/Libs/Const";

jest.mock('react-native-orientation', () => require.requireActual('../../__mocks__/react-native-orientation').default);

describe('Screen >>> Draft', () => {
    const initialState = {
            Cases: {
                draft: {
                    response: [
                        {
                            currentUser: {
                                userId: "00001"
                            }
                        },
                        {
                            currentUser: {
                                userId: "00002"
                            }
                        }
                    ]
                },
                users: [
                    {
                        userId: "00001",
                        userPhoto: "picture1"
                    },
                    {
                        userId: "00002",
                        userPhoto: "picture2"
                    }
                ]
            },
            Screens: {
                All: {
                    disabledItemList: false
                }
            },
            Net: {
                isConnected: true
            }
        },
        requestCasesDraft = jest.fn(),
        userInfoRequest = jest.fn(),
        caseOffline = jest.fn(),
        routeName = jest.fn(),
        reset = jest.fn(),
        navigation = {
            dispatch: jest.fn(),
            setParams: jest.fn(),
            state: {
                routeName: CASE.DRAFT,
                params:{
                    searchInput: null
                }
            }
        },
        mockStore = configureStore();

    let store,
        container,
        viewDraft,
        viewDraftOff;

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(
            <Draft
                requestCasesDraft={requestCasesDraft}
                userInfoRequest={userInfoRequest}
                caseOffline={caseOffline}
                routeName={routeName}
                reset={reset}
                store={store}
            />
        );
        viewDraft = shallow(
            <DraftScreen
                navigation={navigation}
                isConnected={true}
                requestCasesDraft={requestCasesDraft}
                userInfoRequest={userInfoRequest}
                reset={reset}
                routeName={routeName}
                store={store}
            />
        );
        viewDraftOff = shallow(<DraftScreen navigation={navigation} isConnected={false} routeName={routeName} />);
    });

    afterAll(() => {
        RealmObject.close();
    });

    /**
     * Render test
     */
    it('Render draft correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('mapStateToProps', () => {
        expect(container.props('children').requestCasesDraft()).toMatchSnapshot();
        expect(container.props('children').userInfoRequest()).toMatchSnapshot();
        expect(container.props('children').caseOffline()).toMatchSnapshot();
        expect(container.props('children').routeName()).toMatchSnapshot();
        expect(container.props('children').reset()).toMatchSnapshot();
    });

    it('Check static navigationOptions', () => {
        expect(DraftScreen.navigationOptions({navigation}).title)
            .toEqual("menu_drawer_main_menu_item_draft");
        expect(DraftScreen.navigationOptions({navigation}).headerTintColor)
            .toEqual("#FFF");
        expect(DraftScreen.navigationOptions({navigation}).headerStyle)
            .toEqual({backgroundColor: "#1F6CBC"});
        expect(DraftScreen.navigationOptions({navigation}).headerRight).toMatchSnapshot();
    });

    it("Should call navigationOptions method", () => {
        expect(viewDraft.instance().navigationOptions).toBeUndefined();
    });

    it("Props coverage cases", () => {
        expect(container.prop("cases")).toEqual(initialState.Cases);
    });

    it("Props coverage disableItemList", () => {
        expect(container.prop("disableItemList")).toEqual(initialState.Screens.All.disableItemList);
    });

    it("Props coverage componentWillReceiveProps", () => {
        viewDraft.instance().componentWillReceiveProps({
            cases: {
                draft: initialState.Cases.draft
            },
            forceRefresh: true,
            list: CASE.DRAFT,
            disableItemList: initialState.Screens.All.disableItemList
        });
        expect(viewDraft.instance().state.draftRequested).toEqual(false);
        viewDraft.instance().componentWillReceiveProps({
            cases: {
                users: initialState.Cases.users
            },
            list: CASE.DRAFT,
            disableItemList: initialState.Screens.All.disableItemList
        });
        viewDraft.instance().setState({usersRequested: false});
        viewDraft.instance().componentWillReceiveProps({
            cases: {
                draft: {response: []},
                users: initialState.Cases.users
            },
            list: CASE.DRAFT,
            disableItemList: initialState.Screens.All.disableItemList
        });
        expect(viewDraft.instance().state.usersRequested).toEqual(true);
    });

    it("Should call onRefresh method", () => {
        expect(viewDraft.instance().onRefresh()).toMatchSnapshot();
        expect(viewDraft.instance().onRefresh({nativeEvent: {text: "abc"}})).toMatchSnapshot();
        expect(viewDraftOff.instance().onRefresh()).toMatchSnapshot();
    });

    it("Should call changeLayout method", () => {
        viewDraft.instance().changeLayout();
        expect(viewDraft.instance().state.middleHeight).toEqual(617);
    });

    it("Should call goToNewCases method", () => {
        expect(viewDraft.instance().goToNewCases()).toMatchSnapshot();
    });

    it("Should call refresh method of FlatList component", () => {
        const flatList = viewDraft.props("children").children[0];
        expect(flatList.props.renderItem({})).toMatchSnapshot();
        expect(flatList.props.onRefresh()).toMatchSnapshot();
    });

    it("Should call refresh method of FAB component", () => {
        const floatActionButton = viewDraft.props("children").children[1];
        expect(floatActionButton.props.onClickAction()).toMatchSnapshot();
    });
});
