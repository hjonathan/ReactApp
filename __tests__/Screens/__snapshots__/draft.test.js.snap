// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Draft Check static navigationOptions 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "padding": 8,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="search"
    size={12}
    style={
      Object {
        "color": "#FFF",
        "fontSize": 24,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Screen >>> Draft Render draft correctly 1`] = `
<Draft
  caseOffline={[Function]}
  cases={
    Object {
      "draft": Object {
        "response": Array [
          Object {
            "currentUser": Object {
              "userId": "00001",
            },
          },
          Object {
            "currentUser": Object {
              "userId": "00002",
            },
          },
        ],
      },
      "users": Array [
        Object {
          "userId": "00001",
          "userPhoto": "picture1",
        },
        Object {
          "userId": "00002",
          "userPhoto": "picture2",
        },
      ],
    }
  }
  forceRefresh={false}
  isConnected={true}
  list="Draft"
  requestCasesDraft={[Function]}
  reset={[Function]}
  routeName={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userInfoRequest={[Function]}
/>
`;

exports[`Screen >>> Draft Should call goToNewCases method 1`] = `undefined`;

exports[`Screen >>> Draft Should call onRefresh method 1`] = `undefined`;

exports[`Screen >>> Draft Should call onRefresh method 2`] = `undefined`;

exports[`Screen >>> Draft Should call onRefresh method 3`] = `undefined`;

exports[`Screen >>> Draft Should call refresh method of FAB component 1`] = `undefined`;

exports[`Screen >>> Draft Should call refresh method of FlatList component 1`] = `
<ItemList
  isConnected={true}
  parent={
    Object {
      "isConnected": true,
      "navigation": Object {
        "dispatch": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "routeName": "NewCases",
                "type": "Navigation/NAVIGATE",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "setParams": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "state": Object {
          "params": Object {
            "searchInput": null,
          },
          "routeName": "Draft",
        },
      },
      "requestCasesDraft": [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {
              "search": "",
            },
          ],
          Array [
            Object {},
          ],
          Array [
            Object {
              "search": "",
            },
          ],
          Array [
            Object {
              "search": "abc",
            },
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "reset": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "routeName": [MockFunction] {
        "calls": Array [
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "userInfoRequest": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "size": "1",
              "user": Object {
                "ids": Array [
                  "00001",
                  "00002",
                ],
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    }
  }
/>
`;

exports[`Screen >>> Draft Should call refresh method of FlatList component 2`] = `undefined`;

exports[`Screen >>> Draft mapStateToProps 1`] = `
Object {
  "type": "cases/draft/request",
}
`;

exports[`Screen >>> Draft mapStateToProps 2`] = `
Object {
  "type": "cases/userInfo/request",
}
`;

exports[`Screen >>> Draft mapStateToProps 3`] = `
Object {
  "type": "net/case/list",
}
`;

exports[`Screen >>> Draft mapStateToProps 4`] = `
Object {
  "type": "cases/routeName",
}
`;

exports[`Screen >>> Draft mapStateToProps 5`] = `
Object {
  "type": "cases/reset",
}
`;
